lib/src/convert_es6_proxy.dart

```diff
@@ -0,0 +1,59 @@
+@HtmlImport('convert_es6_proxy.html')
+library polymer_interop.lib.src.convert_es_proxy;
+
+import 'package:web_components/web_components.dart' show HtmlImport;
+
+import 'dart:js';
+
+import 'package:polymer_interop/polymer_interop.dart';
+export 'package:polymer_interop/polymer_interop.dart'
+    show convertToDart, convertToJs;
+
+final JsObject _polymerInteropDartES6 = context['Polymer']['PolymerInterop']['ES6'];
+
+final JsFunction createES6JsProxyForArray = _initES6ListProxySupport();
+
+final JsObject _Unsupported = _polymerInteropDartES6['Unsupported'];
+
+/// Hooks for ES6 Proxies
+///
+/// Hooks for getting and setting properties and methods from and to dart objects.
+///
+JsFunction _initES6ListProxySupport() {
+  <String, Function>{
+    '_dartArrayGet': (List instance, index) =>
+        (index is num) ? convertToJs(instance[index]) : _Unsupported,
```

Comment:
I still have to figure out this. An Array in JS-lang is sometime decorated with other properties that are not number. Dart interop for example adds some hiddent properties to store the dart equivalent object and Polymer too will if I remember.

But actually _dartArrayGet should be used only when index is an integer so it's right here to seti it as an int.